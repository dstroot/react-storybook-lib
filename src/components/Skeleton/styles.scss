.skeleton {
  display: inline-block;
  width: 100%;
  background: linear-gradient(-90deg, #f0f0f0 0%, #f9f9f9 50%, #f0f0f0 100%);
  background-size: 400%;
  animation: pulse 1.2s ease-in-out infinite;
  border-radius: 5px;
  border-style: solid;
  border-color: white;
  border-width: 1px;
  margin: -1px;
  &::before {
    content: '\00a0';
  }
}

.skeleton-round {
  display: inline-block;
  width: 100%;
  background: linear-gradient(-90deg, #f0f0f0 0%, #f9f9f9 50%, #f0f0f0 100%);
  background-size: 400%;
  animation: pulse 1.2s ease-in-out infinite;
  border-radius: 50%;
  border-style: solid;
  border-color: white;
  border-width: 1px;
  margin: -1px;
  &::before {
    content: '\00a0';
  }
}

@keyframes pulse {
  0% {
    background-position: 0% 0%;
  }
  100% {
    background-position: -135% 0%;
  }
}
