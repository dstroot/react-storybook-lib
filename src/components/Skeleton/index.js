import React from 'react';
// import styled, { keyframes } from 'styled-components';

// styles
import './styles.scss';

/**
 * https://medium.com/octopus-wealth/skeleton-loading-pages-with-react-5a931f12677b
 * https://stackoverflow.com/questions/51338631/react-how-to-specify-animation-keyframes-and-classes-locally
 */

// const SSkeletonPulse = styled.div`
//   display: inline-block;
//   height: 100%;
//   width: 100%;
//   background: linear-gradient(-90deg, #f0f0f0 0%, #f8f8f8 50%, #f0f0f0 100%);
//   background-size: 400%;
//   animation: pulse 1.2s ease-in-out infinite;
//   @keyframes pulse {
//     0% {
//       background-position: 0% 0%;
//     }
//     100% {
//       background-position: -135% 0%;
//     }
//   }
// `;

// const SSkeletonLine = styled(SSkeletonPulse)`
//   width: 5.5em;
//   border-radius: 5px;

//   &::before {
//     content: '\00a0';
//   }
// `;

// const Skeleton = () => <SSkeletonLine />;

const Skeleton = () => <div className="skeleton" />;

export default Skeleton;
