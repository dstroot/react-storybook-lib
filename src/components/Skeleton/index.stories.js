import React from 'react';
import { storiesOf } from '@storybook/react';
import Skeleton from '../Skeleton';

const testStyle = {
  height: '50px',
  width: '100%',
};

storiesOf('Misc|Skeleton', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!
      
      ~~~js
      console.log("hello");
      ~~~
      `,
    },
  })
  .add('show Skeleton', () => (
    <div className="container">
      <div className="row">
        <div className="col-6">
          <div style={testStyle}>
            <Skeleton />
          </div>
        </div>
        <div className="col-6">
          <div style={testStyle}>
            <Skeleton />
          </div>
        </div>
      </div>
    </div>
  ));
