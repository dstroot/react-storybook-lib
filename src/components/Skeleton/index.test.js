import React from 'react';
import { create } from 'react-test-renderer';

import Skeleton from '../Skeleton';

describe('Skeleton', () => {
  it('it should render', () => {
    let tree = create(<Skeleton />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
