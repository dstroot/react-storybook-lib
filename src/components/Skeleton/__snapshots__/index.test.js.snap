// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton it should render 1`] = `
<div
  className="skeleton"
/>
`;
